src/modules/DASDController.rb

```diff
@@ -851,6 +791,106 @@ def report_error(headline, message, details)
         Yast2::Popup.show(message, headline: headline, details: details)
       end
     end
+
+    # Activates a disk if its not activated
+    #
+    # When the disk is already activated, it returns '8' if the
+    # disk is unformatted or '0' otherwise. The idea is to mimic
+    # the same API that ActivateDisk.
+    #
+    # @return [Boolean] Returns an error code (8 means 'unformatted').
+    def activate_disk_if_needed(channel, diag)
+      disk = find_disks.find { |d| d["channel"] == channel }
+      if disk && active_disk?(disk)
+        log.info "Disk #{channel} is already active. Skipping the activation."
+        return disk["formatted"] ? 0 : 8
+      end
+      ActivateDisk(channel, diag) unless disk && active_disk?(disk)
+    end
+
+    # Determines whether the disk is activated or not
+    #
+    # @param disk [Hash]
+    # @return [Boolean]
+    def active_disk?(disk)
+      io = disk.fetch("resource", {}).fetch("io", []).first
+      !!(io && io["active"])
+    end
+
+    # Returns the DASD disks
+    #
+    # Probes and returns the found DASD disks ordered by channel.
+    # It caches the found disks.
+    #
+    # @param force_probing [Boolean] Ignore the cached values and probes again.
+    # @return [Array<Hash>] Found DASD disks
+    def find_disks(force_probing = false)
```

Comment:
for new API please do not use boolean parameters if it is not really straighforward what it means. And this is not case. If I see `find_disks(false)` and find_disks(true)` without checkout documentation, I have no clue what that true/false means.
I prefer named parameters so I see `find_disks(force_probing: true)` and `find_disks(force_probing: false) when I immediatelly see what it means.